airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/core/operation/executor/DefaultReadOperationExecutor.kt

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.core.operation.executor
+
+import io.airbyte.cdk.core.context.env.ConnectorConfigurationPropertySource
+import io.airbyte.cdk.core.operation.OperationExecutionException
+import io.airbyte.cdk.core.util.ShutdownUtils
+import io.airbyte.commons.util.AutoCloseableIterator
+import io.airbyte.protocol.models.AirbyteStreamNameNamespacePair
+import io.airbyte.protocol.models.v0.AirbyteMessage
+import io.github.oshai.kotlinlogging.KotlinLogging
+import io.micronaut.context.annotation.Requires
+import jakarta.inject.Named
+import jakarta.inject.Singleton
+import java.util.Optional
+import java.util.concurrent.TimeUnit
+import java.util.function.Consumer
+
+private val logger = KotlinLogging.logger {}
+
+@Singleton
+@Named("readOperationExecutor")
+@Requires(
+    property = ConnectorConfigurationPropertySource.CONNECTOR_OPERATION,
+    value = "read",
+)
+@Requires(env = ["source"])
+class DefaultReadOperationExecutor(
+    private val messageIterator: Optional<AutoCloseableIterator<AirbyteMessage>>,
+    @Named("outputRecordCollector") private val outputRecordCollector: Consumer<AirbyteMessage>,
+    private val shutdownUtils: ShutdownUtils,
+) : OperationExecutor {
+    override fun execute(): Result<AirbyteMessage?> {
+        logger.info { "Using default read operation executor." }
+        try {
+            if (messageIterator.isPresent) {
+                val iterator = messageIterator.get()
+                try {
+                    iterator.use {
+                        iterator.airbyteStream.ifPresent { s: AirbyteStreamNameNamespacePair? ->
+                            logger.debug { "Producing messages for stream $s..." }
+                        }
+                        iterator.forEachRemaining(outputRecordCollector)
+                        iterator.airbyteStream.ifPresent { s: AirbyteStreamNameNamespacePair? ->
+                            logger.debug { "Finished producing messages for stream $s..." }
+                        }
+                    }
```

Comment:
Is `use` the right scoped method to use? I'm still not very familiar with these but I wouldn't expect `iterator` to be referenced directly in the body of whatever scoped method is appropriate (instead, relying on the implicit `this`).

Reply:
@postamar This is generated code from the Java -> Kotlin conversion, so I will take another pass to see if I can clean this up.

Reply:
@postamar I take this back.  The use of `use` is to ensure that the iterator is automatically closed when it is done being used.